cfn-resources/stream-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	updateRequest := &admin.StreamsDataProcessRegion{
+		CloudProvider: *currentModel.DataProcessRegion.CloudProvider,
+		Region:        *currentModel.DataProcessRegion.Region,
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	_, resp, err := atlasV2.StreamsApi.UpdateStreamInstance(context.Background(), *currentModel.ProjectId, *currentModel.InstanceName, updateRequest).Execute()
+	if err != nil {
+		return handleError(resp, constants.UPDATE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update success",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(DeleteRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	_, resp, err := atlasV2.StreamsApi.DeleteStreamInstance(context.Background(), *currentModel.ProjectId, *currentModel.InstanceName).Execute()
+	if err != nil {
+		return handleError(resp, constants.DELETE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete success"}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(ListRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	accumulatedStreamInstances := make([]admin.StreamsTenant, 0)
+	pageNum := 1
+	for ok := true; ok; {
+		listStreamInstancesRequest := atlasV2.StreamsApi.ListStreamInstances(context.Background(), *currentModel.ProjectId)
+		listStreamInstancesRequest.PageNum(pageNum)
+		listStreamInstancesRequest.ItemsPerPage(100)
+		streamInstances, resp, err := listStreamInstancesRequest.Execute()
+		if err != nil {
+			return handleError(resp, constants.LIST, err)
+		}
+		accumulatedStreamInstances = append(accumulatedStreamInstances, streamInstances.GetResults()...)
+		ok = streamInstances.GetTotalCount() > len(accumulatedStreamInstances)
```

Comment:
not understanding this. 

`totalCount` is the `Number of documents returned in this response if includeCount query param is true.`

so it seems it returns only the number of documents of the specific API call. How can this condition work to determine if all pages have been checked?

Reply:
Looking at the backend, totalCount is always returned(if number of results is greater than zero) and it is the number of total results, not the amount returned in that specific request. Probably the description is not correct.
[handlePagination ](https://github.com/10gen/mms/blob/834aeda6f8337dcba3b3354877e12d069672a4db/server/src/main/com/xgen/cloud/common/res/_public/base/ApiBaseResource.java#L203) method
[handlePaginationInternal](https://github.com/10gen/mms/blob/834aeda6f8337dcba3b3354877e12d069672a4db/server/src/main/com/xgen/cloud/common/res/_public/base/ApiBaseResource.java#L227) method

Reply:
can you follow-up separately (not a blocker) with the API team and ask them clarifications?

Reply:
Fix for docs will go live later this week